lib/index.js

```diff
@@ -85,25 +95,82 @@ Vero.prototype.track = function(payload, fn) {
  * @api public
  */
 
-Vero.prototype.alias = request('/users/reidentify', 'put');
+Vero.prototype.alias = request('alias', '/users/reidentify', 'put');
+
+/**
+ * Add or Remove Tags.
+ *
+ * http://developers.getvero.com/?bash#tags
+ *
+ * @param {Facade} msg
+ * @param {Object} tags
+ * @param {Function} fn
+ * @api public
+ */
+
+Vero.prototype.addOrRemoveTags = function(msg, tags, fn) {
+  // Check to ensure tags object is properly formatted.
+  var err;
+  if (Object.prototype.toString.call(tags) !== '[object Object]') {
+    err = new Error('Tag option must be passed as an Object.');
+  }
+
+  if (!tags.action || tags.action !== 'add' && tags.action !== 'remove') {
+    err = new Error('Tags action value must be either \'add\' or \'remove\'');
+  }
+
+  if (!tags.values || !Array.isArray(tags.values)) {
+    err = new Error('Could not detect tag values. Tag values must be passed as array of strings.');
+  }
+
+  if (err instanceof Error) {
+    err.status = 400;
+    return fn(err);
+  }
+
+  var payload = mapper.addOrRemoveTags(msg, tags, this.settings);
+  var path = '/users/tags/edit';
+
+  this
+    .request('put', path)
+    .set('Accept', 'application/json')
+    .type('json')
+    .send(payload)
+    .end(fn);
+};
 
 /**
  * Request.
  *
+ * @param {String} integrationMethod
  * @param {String} path
- * @param {String} method
+ * @param {String} requestMethod
  * @return {Function}
  * @api private
  */
 
-function request(path, method){
-  method = method || 'post';
-  return function(payload, fn){
-    return this
-      .request(method, path)
+function request(integrationMethod, path, requestMethod) {
+  requestMethod = requestMethod || 'post';
+  return function(msg, fn) {
+    var payload = mapper[integrationMethod](msg, this.settings);
+    var self = this;
+    return self
+      .request(requestMethod, path)
       .set('Accept', 'application/json')
       .type('json')
       .send(payload)
-      .end(this.handle(fn));
+      .then(function(res) {
+        var tags = checkForTags(msg);
+        if (tags) return self.addOrRemoveTags(msg, tags, fn);
+        return fn(null, res);
+      }, fn);
   };
 }
+
+function checkForTags(msg) {
```

Comment:
Is this necessary functionality for the request or should this live in the mapper?